localstack-core/localstack/services/apigateway/next_gen/execute_api/integrations/aws.py

```diff
@@ -37,3 +87,173 @@ class RestApiAwsProxyIntegration(RestApiIntegration):
+
+        if method != HTTPMethod.POST:
+            raise IntegrationFailureError("Internal server error")
+
+        input_event = self.create_lambda_input_event(context)
+
+        function_arn = get_lambda_function_arn_from_invocation_uri(integration["uri"])
+        source_arn = get_source_arn(context)
+        # TODO: properly get the value out/validate it?
+        raw_headers = context.invocation_request["raw_headers"]
+        is_invocation_asynchronous = raw_headers.get("X-Amz-Invocation-Type") == "'Event'"
+
+        try:
+            lambda_payload = self.call_lambda(
+                function_arn=function_arn,
+                event=to_bytes(json.dumps(input_event)),
+                source_arn=source_arn,
+                credentials=integration.get("credentials"),
+                asynchronous=is_invocation_asynchronous,
+            )
+        except ClientError as e:
+            # TODO: more data from the error message?
+            LOG.warning(
+                "Exception during integration invocation: '%s'",
+                e,
+            )
+            raise IntegrationFailureError("Internal server error", status_code=502) from e
+        except Exception as e:
+            LOG.warning(
+                "Unexpected exception during integration invocation: '%s'",
+                e,
+            )
+            raise IntegrationFailureError("Internal server error", status_code=502) from e
+
+        lambda_response = self.parse_lambda_response(lambda_payload)
+
+        headers = merge_dicts(
+            {"Content-Type": "application/json"},
+            lambda_response.get("headers") or {},
+            lambda_response.get("multiValueHeaders") or {},
+        )
+        # TODO: look into remapped headers (like HTTP_PROXY and integration response handlers)
+
+        response = Response(
+            headers=Headers(headers),
+            response=lambda_response.get("body") or "",
+            status=lambda_response.get("statusCode") or 200,
+        )
+
+        return response
+
+    @staticmethod
+    def call_lambda(
+        function_arn: str,
+        event: bytes,
+        source_arn: str,
+        credentials: str = None,
+        asynchronous: bool = False,
+    ) -> bytes:
+        lambda_client = get_service_factory(
+            region_name=extract_region_from_arn(function_arn),
+            role_arn=credentials,
+        ).lambda_
+        inv_result = lambda_client.request_metadata(
+            service_principal=ServicePrincipal.apigateway,
+            source_arn=source_arn,
+        ).invoke(
+            FunctionName=function_arn,
+            Payload=event,
+            InvocationType="Event" if asynchronous else "RequestResponse",
+        )
+        if payload := inv_result.get("Payload"):
+            return payload.read()
+        return b""
+
+    def parse_lambda_response(self, payload: bytes) -> LambdaProxyResponse:
+        try:
+            lambda_response = json.loads(payload)
```

Comment:
Is there still a weird edge case where a lambda returning `"[]"` would lead to a `TypeError` down the line?